### PR TITLE
Add new Lambda template for the AWS Message Processing Framework

### PR DESCRIPTION
**Pull request by ashovlin**
*Issue #, if available:* DOTNET-7413

*Description of changes:* Adds a Lambda blueprint to use with the [AWS Message Processing Framework for .NET](https://github.com/awslabs/aws-dotnet-messaging).

It consists of two Lambda functions - one that publishes to a queue and one that consumes messages from the same queue.

I've tested `dotnet msbuild .\buildtools\build.proj -t:test-blueprints-dotnew`, still doing some manual end-to-end testing.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/BlueprintBaseName.1.csproj
Comment (ashovlin) [inline Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/BlueprintBaseName.1.csproj:18]: Planning to come back and update this before merging once we drop the prerelease tag.

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Functions.cs
Comment (normj) [inline Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Functions.cs:28]: You could be modern and use [primary constructors](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/instance-constructors#primary-constructors).

No big deal if you don't want to and save that effort to another time of going through the other annotations based templates.

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Functions.cs
Comment (normj) [inline Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Functions.cs:23]: Not sure the context of adding the following to this constructor.
```
When invoked in a Lambda environment the AWS credentials will come from the IAM role associated 
with the function and the AWS region will be set to the region the Lambda function is executed in.
```

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Functions.cs
Comment (normj) [inline Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Functions.cs:40]: The `FromBody` is for APIGateway based functions to mean the HTTP request body. I pretty sure in this context it is being ignored. Although it might be a better example to turn this into an API Gateway based Lambda function.

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Functions.cs
Comment (normj) [inline Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Functions.cs:47]: I would add the `ILambdaContext` to the parameters and use that. It will be even more important when we add structured logging support.

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Functions.cs
Comment (normj) [inline Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Functions.cs:69]: Put in a separate file.

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Functions.cs
Comment (normj) [inline Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Functions.cs:77]: Inject the ILambdaContext and write some log message in particular why it failed.

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Readme.md
Comment (normj) [inline Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Readme.md:7]: The `serverless.template` is missing in the PR.

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Startup.cs
Comment (normj) [inline Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Startup.cs:34]: Maybe add a comment about adding the OTel package? Or you could add something about that in the README.

**Review by normj — CHANGES_REQUESTED** [review]
The PR is missing the `serverless.template`.

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Startup.cs
Comment (ashovlin) [reply Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Startup.cs:34]: Added to README in [4ad2937](https://github.com/aws/aws-lambda-dotnet/pull/1707/commits/4ad293784c1044b65deecb54b426b6520891b39a)

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Readme.md
Comment (ashovlin) [reply Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Readme.md]: Oops, added in [4ad2937](https://github.com/aws/aws-lambda-dotnet/pull/1707/commits/4ad293784c1044b65deecb54b426b6520891b39a)

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Functions.cs
Comment (ashovlin) [reply Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Functions.cs]: Done via [4ad2937](https://github.com/aws/aws-lambda-dotnet/pull/1707/commits/4ad293784c1044b65deecb54b426b6520891b39a)

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Functions.cs
Comment (ashovlin) [reply Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Functions.cs]: Moved via [4ad2937](https://github.com/aws/aws-lambda-dotnet/pull/1707/commits/4ad293784c1044b65deecb54b426b6520891b39a)

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Functions.cs
Comment (ashovlin) [reply Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Functions.cs]: Removed in [4ad2937](https://github.com/aws/aws-lambda-dotnet/pull/1707/commits/4ad293784c1044b65deecb54b426b6520891b39a).

I didn't switch to APIG yet, was leaning towards keeping it simple. Though agree that it's more realistic, so let me know if you or others feel strongly.

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Functions.cs
Comment (ashovlin) [reply Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Functions.cs]: It's in the plain SQS template which I started from, removed from here in [4ad2937](https://github.com/aws/aws-lambda-dotnet/pull/1707/commits/4ad293784c1044b65deecb54b426b6520891b39a).

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/.lambda-test-tool/SavedRequests/Handler Sampler Request.json
Comment (ashovlin) [reply Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/.lambda-test-tool/SavedRequests/Handler Sampler Request.json]: Removed spaces, agree with not worrying about cross-platform/side effects/etc. of spaces.

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/blueprint-manifest.json
Comment (ashovlin) [reply Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/blueprint-manifest.json]: Updated via [4ad2937](https://github.com/aws/aws-lambda-dotnet/pull/1707/commits/4ad293784c1044b65deecb54b426b6520891b39a)

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/blueprint-manifest.json
Comment (ashovlin) [reply Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/blueprint-manifest.json]: Updated via [4ad2937](https://github.com/aws/aws-lambda-dotnet/pull/1707/commits/4ad293784c1044b65deecb54b426b6520891b39a)

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/blueprint-manifest.json
Comment (ashovlin) [reply Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/blueprint-manifest.json]: Updated via [4ad2937](https://github.com/aws/aws-lambda-dotnet/pull/1707/commits/4ad293784c1044b65deecb54b426b6520891b39a)

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/blueprint-manifest.json
Comment (ashovlin) [reply Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/blueprint-manifest.json]: Updated via [4ad2937](https://github.com/aws/aws-lambda-dotnet/pull/1707/commits/4ad293784c1044b65deecb54b426b6520891b39a)

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Readme.md
Comment (philasmar) [inline Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Readme.md:4]: nit: drop "a" before "two"

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/serverless.template
Comment (philasmar) [inline Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/serverless.template:17]: Should this be 512?

**Review by philasmar — CHANGES_REQUESTED** [review]

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/serverless.template
Comment (ashovlin) [reply Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/serverless.template:47]: We still want to update Lambda Annotations itself to default to 512. 

I could override via the attributes in this specific template, but would rather just update once centrally.

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/serverless.template
Comment (ashovlin) [reply Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/serverless.template:17]: This is being set by Lambda Annotations, which we want to update its default value to 512.

I could override this specific template via the attributes, but would rather just update once centrally.

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/blueprint-manifest.json
Comment (ashovlin) [reply Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/blueprint-manifest.json:2]: image

Norm recommended this above via https://github.com/aws/aws-lambda-dotnet/pull/1707#discussion_r1538337074, we want to fit in the VS UI.

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Functions.cs
Comment (ashovlin) [reply Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Functions.cs]: Updated in https://github.com/aws/aws-lambda-dotnet/pull/1707/commits/f67478189c12ca94cc6ae866f8e95490e7f61360

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/blueprint-manifest.json
Comment (ashovlin) [reply Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/blueprint-manifest.json]: Updated since it does fit in the UI
image

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/blueprint-manifest.json
Comment (ashovlin) [reply Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/blueprint-manifest.json]: I changed to just _"Messaging"_ since the UI is still relatively small.

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/GreetingMessageHandler.cs
Comment (ashovlin) [reply Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/GreetingMessageHandler.cs]: Done in https://github.com/aws/aws-lambda-dotnet/pull/1707/commits/f67478189c12ca94cc6ae866f8e95490e7f61360

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Readme.md
Comment (ashovlin) [reply Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Readme.md]: Fixed in https://github.com/aws/aws-lambda-dotnet/pull/1707/commits/f67478189c12ca94cc6ae866f8e95490e7f61360

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Readme.md
Comment (ashovlin) [reply Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Readme.md]: Fixed in https://github.com/aws/aws-lambda-dotnet/pull/1707/commits/f67478189c12ca94cc6ae866f8e95490e7f61360

**Review by philasmar — APPROVED** [review]

**Review by normj — CHANGES_REQUESTED** [review]

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Functions.cs
Comment (ashovlin) [reply Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Functions.cs:58]: Switched in https://github.com/aws/aws-lambda-dotnet/pull/1707/commits/76e5c518c74d1010e2281d9c8fb0a68f401dc964

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/serverless.template
Comment (ashovlin) [reply Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/serverless.template:37]: Added in https://github.com/aws/aws-lambda-dotnet/pull/1707/commits/76e5c518c74d1010e2281d9c8fb0a68f401dc964

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Functions.cs
Comment (normj) [reply Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Functions.cs:58]: You have to change the return to `Task<SQSBatchResponse>`

**Review by normj — CHANGES_REQUESTED** [review]

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Functions.cs
Comment (ashovlin) [reply Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/Functions.cs]: Oops, fixed in https://github.com/aws/aws-lambda-dotnet/pull/1707/commits/85033a007d36017a48604f2c2e0fc457e67ddb8e

File: Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/aws-lambda-tools-defaults.json
Comment (normj) [inline Blueprints/BlueprintDefinitions/vs2022/MessageProcessingFramework/template/src/BlueprintBaseName.1/aws-lambda-tools-defaults.json:11]: You don't need the `function-runtime`, `function-memory-size` and `function-timeout` because they are set in the CloudFormation template.

**Review by normj — APPROVED** [review]
One small nit but you don't need my approval again for that change.

**Pull request merged**
